File: ts-solutions/tests/1207-unique-occurrences.test.ts
import { uniqueOccurrences } from '../src/1207-unique-occurrences'

type UniqueOccurrencesTest = {
	arr: number[]
	want: boolean
}

const uniqueOccurrencesTests: UniqueOccurrencesTest[] = [
	{ arr: [1, 2, 2, 1, 1, 3], want: true },
	{ arr: [1, 2], want: false },
	{ arr: [-3, 0, 1, -3, 1, 1, 1, -3, 10, 0], want: true },
]

describe(`uniqueOccurrences`, () => {
	test.each(uniqueOccurrencesTests)(
		`uniqueOccurrences($arr) => $want`,
		({ arr, want }) => {
			expect(uniqueOccurrences(arr)).toBe(want)
		}
	)
})

File: go-solutions/1207_unique_occurrences_test.go
package leetcode

import (
	"fmt"
	"testing"
)

type uniqueOccurrencesTest struct {
	arr  []int
	want bool
}

var uniqueOccurrencesTests = []uniqueOccurrencesTest{
	{[]int{1, 2, 2, 1, 1, 3}, true},
	{[]int{1, 2}, false},
	{[]int{-3, 0, 1, -3, 1, 1, 1, -3, 10, 0}, true},
}

func TestUniqueOccurrence(t *testing.T) {
	for _, test := range uniqueOccurrencesTests {
		t.Run(fmt.Sprintf("UniqueOccurrences(%v) => %v", test.arr, test.want), func(t *testing.T) {
			got := UniqueOccurrences(test.arr)
			if got != test.want {
				t.Errorf("got %v, want %v", got, test.want)
			}
		})
	}
}

func BenchmarkUniqueOccurrences(b *testing.B) {
	for i := 0; i < b.N; i++ {
		for _, test := range uniqueOccurrencesTests {
			UniqueOccurrences(test.arr)
		}
	}
}
